processor/k8sattributesprocessor/e2e_test.go

```diff
@@ -175,10 +176,10 @@ func scanSpanForAttributes(t *testing.T, expectedService string, kvs map[string]
 			if service.AsString() != expectedService {
 				continue
 			}
-
-			assert.NoError(t, resourceHasAttributes(resourceSpans.Resource(), kvs))
+			err = resourceHasAttributes(resourceSpans.Resource(), kvs)
```

Comment:
should this check if the return from resourceHasAttribute is `nil` before assigning it? Otherwise I would expect this test to continue being flaky

Reply:
I wrote like this, because, assume that there exist N lines of trace data,

the successful test case may be like following 2 

- the first M lines do not have the attributes appended, but the following N-M lines do have
- all the N lines do have the attributes appended

In those two cases, the `err` will all finally be nil.

the failed case must be 

- all the N lines  do not have the attributes appended 

so the `err` will be not nil.

I am not sure I explain my thinking clearly. :)

Reply:
It's an appropriate remediation for the test flakiness, but it doesn't solve the root cause which is in the k8s attribute processor that doesn't enrich all the traces from the beginning as expected. @fatsheep9146, can you please file an issue so someone can follow up later on?

Reply:
ok, the issue is created. #18892
@dmitryax